nautobot/extras/management/commands/fix_custom_fields.py

```diff
@@ -0,0 +1,39 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from nautobot.extras.utils import FeatureQuery
+
+
+class Command(BaseCommand):
+    help = "Reset custom field attachments"
+
+    def handle(self, *args, **kwargs):
+        """Run through all objects and ensure they are associated with the correct custom fields."""
+        content_types = ContentType.objects.filter(FeatureQuery("custom_fields").get_query())
+
+        for content_type in content_types:
+            self.stdout.write(self.style.SUCCESS(f"Processing ContentType {content_type}"))
+            model = content_type.model_class()
+            custom_fields_for_content_type = content_type.custom_fields.all()
+            custom_field_names_for_content_type = [cf.name for cf in custom_fields_for_content_type]
+            with transaction.atomic():
+                for obj in model.objects.all():
```

Comment:
Not sure if it'd make output too verbose, but maybe add a `Processing {content_type} {obj}` message here? Or could even get fancy and pull in `tqdm` as a dependency to have a progress bar per content-type, similar to nautobot-netbox-importer? Maybe overkill, just thinking. :-)

Reply:
@glennmatthews adding a message for every object could get way too verbose.  I think tqdm is a little overkill for this command.  Unless you really want it then I can spend some time and figure it out.